### PR TITLE
Fix bin/setup and bundler conflict

### PR DESCRIPTION
**Pull request by Fryguy**

### CODE REVIEW

**Comment by Fryguy** [comment]
Ugh there's a bunch of stuff in the shared setup that reference travis specifically...gonna need to fix that in core first.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -25,13 +25,22 @@ jobs:
...
+    - name: Install system packages
+      run: sudo apt-get install -y libcurl4-openssl-dev
     - name: Set up Ruby
       uses: ruby/setup-ruby@v1
       with:
         ruby-version: ${{ matrix.ruby-version }}
-        bundler-cache: true
     - name: Prepare tests
       run: bin/setup
+    - name: Set up bundler caching
```
Comment (kbrock) [inline .github/workflows/ci.yaml:39]: you are doing this twice?

Comment (Fryguy) [reply .github/workflows/ci.yaml:39]: I'm experimenting in this PR...ignore the man behind the curtain 😆

**Comment by chessbyte** [comment]
@Fryguy CI via GitHub Actions is passing for both 2.6 and 2.7.  What else is needed to get this and the related PR over the finish line?

**Comment by Fryguy** [comment]
Updated the PR after some discussion with @agrare.  I've moved the repo cloning and system packages into a `bin/before_install` script, which is called from GitHub Actions before we install Ruby, and is also called from `bin/setup` to preserve that functionality for users.  This also keeps the cloning in one place which is important for the release scripts at branching/tagging time since they need to also change the cloned core ref. (e.g. changing clone of core to najdorf when we cut the najdorf branch on the provider repo).  The `bin/before` script is a shell script since we don't have Ruby yet.

**Review by agrare — APPROVED** [review]

**Comment by Fryguy** [comment]
One weird thing is that technically we bundle install twice...this can easily be handled in core, but it only incurs ~15s of overhead, so I'm going to solve that later once we are all back to green

**Pull request merged**